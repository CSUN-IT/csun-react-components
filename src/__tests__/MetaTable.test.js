import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import {
  MetaTable,
  MetaTableHead,
  MetaTableRow,
  MetaTableCell,
  MetaTableHeaderCell,
  MetaTableBody,
} from '../../src';

configure({ adapter: new Adapter() });

describe('MetaTable', () => {
  it('Should be defined', () => {
    expect(MetaTable).toBeDefined();
  });

  it('Renders component correctly', () => {
    const tree = shallow(
      <MetaTable>
        <MetaTableHead>
          <MetaTableRow>
            <MetaTableHeaderCell>Hello</MetaTableHeaderCell>
            <MetaTableHeaderCell>World</MetaTableHeaderCell>
          </MetaTableRow>
        </MetaTableHead>
        <MetaTableBody>
          <MetaTableRow>
            <MetaTableCell>Foo</MetaTableCell>
            <MetaTableCell>Bar</MetaTableCell>
          </MetaTableRow>
        </MetaTableBody>
      </MetaTable>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Should mount and check table row contains two children', () => {
    const tree = mount(
      <MetaTable>
        <MetaTableHead>
          <MetaTableRow>
            <MetaTableHeaderCell>Hello</MetaTableHeaderCell>
            <MetaTableHeaderCell>World</MetaTableHeaderCell>
          </MetaTableRow>
        </MetaTableHead>
        <MetaTableBody>
          <MetaTableRow>
            <MetaTableCell>Foo</MetaTableCell>
            <MetaTableCell>Bar</MetaTableCell>
          </MetaTableRow>
        </MetaTableBody>
      </MetaTable>,
    );
    expect(tree.find('MetaTableRow').children()).toHaveLength(2);
  });
});
