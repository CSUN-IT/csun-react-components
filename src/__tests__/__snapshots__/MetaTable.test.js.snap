// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaTable Renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MetaTable
    className=""
    header={Array []}
  >
    <MetaTableHead>
      <MetaTableRow>
        <TableHeaderCell>
          Hello
        </TableHeaderCell>
        <TableHeaderCell>
          World
        </TableHeaderCell>
      </MetaTableRow>
    </MetaTableHead>
    <MetaTableBody>
      <MetaTableRow>
        <MetaTableCell>
          Foo
        </MetaTableCell>
        <MetaTableCell>
          Bar
        </MetaTableCell>
      </MetaTableRow>
    </MetaTableBody>
  </MetaTable>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <table
        className="table table--striped table--bordered table--padded table--hover"
      >
        <MetaTableHead>
          <MetaTableRow>
            <TableHeaderCell>
              Hello
            </TableHeaderCell>
            <TableHeaderCell>
              World
            </TableHeaderCell>
          </MetaTableRow>
        </MetaTableHead>
        <MetaTableBody>
          <MetaTableRow>
            <MetaTableCell>
              Foo
            </MetaTableCell>
            <MetaTableCell>
              Bar
            </MetaTableCell>
          </MetaTableRow>
        </MetaTableBody>
      </table>,
      "className": "table-responsive",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <MetaTableHead>
            <MetaTableRow>
              <TableHeaderCell>
                Hello
              </TableHeaderCell>
              <TableHeaderCell>
                World
              </TableHeaderCell>
            </MetaTableRow>
          </MetaTableHead>,
          <MetaTableBody>
            <MetaTableRow>
              <MetaTableCell>
                Foo
              </MetaTableCell>
              <MetaTableCell>
                Bar
              </MetaTableCell>
            </MetaTableRow>
          </MetaTableBody>,
        ],
        "className": "table table--striped table--bordered table--padded table--hover",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <MetaTableRow>
              <TableHeaderCell>
                Hello
              </TableHeaderCell>
              <TableHeaderCell>
                World
              </TableHeaderCell>
            </MetaTableRow>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <TableHeaderCell>
                  Hello
                </TableHeaderCell>,
                <TableHeaderCell>
                  World
                </TableHeaderCell>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Hello",
                },
                "ref": null,
                "rendered": "Hello",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "World",
                },
                "ref": null,
                "rendered": "World",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <MetaTableRow>
              <MetaTableCell>
                Foo
              </MetaTableCell>
              <MetaTableCell>
                Bar
              </MetaTableCell>
            </MetaTableRow>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <MetaTableCell>
                  Foo
                </MetaTableCell>,
                <MetaTableCell>
                  Bar
                </MetaTableCell>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Foo",
                },
                "ref": null,
                "rendered": "Foo",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Bar",
                },
                "ref": null,
                "rendered": "Bar",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "table",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <table
          className="table table--striped table--bordered table--padded table--hover"
        >
          <MetaTableHead>
            <MetaTableRow>
              <TableHeaderCell>
                Hello
              </TableHeaderCell>
              <TableHeaderCell>
                World
              </TableHeaderCell>
            </MetaTableRow>
          </MetaTableHead>
          <MetaTableBody>
            <MetaTableRow>
              <MetaTableCell>
                Foo
              </MetaTableCell>
              <MetaTableCell>
                Bar
              </MetaTableCell>
            </MetaTableRow>
          </MetaTableBody>
        </table>,
        "className": "table-responsive",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <MetaTableHead>
              <MetaTableRow>
                <TableHeaderCell>
                  Hello
                </TableHeaderCell>
                <TableHeaderCell>
                  World
                </TableHeaderCell>
              </MetaTableRow>
            </MetaTableHead>,
            <MetaTableBody>
              <MetaTableRow>
                <MetaTableCell>
                  Foo
                </MetaTableCell>
                <MetaTableCell>
                  Bar
                </MetaTableCell>
              </MetaTableRow>
            </MetaTableBody>,
          ],
          "className": "table table--striped table--bordered table--padded table--hover",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <MetaTableRow>
                <TableHeaderCell>
                  Hello
                </TableHeaderCell>
                <TableHeaderCell>
                  World
                </TableHeaderCell>
              </MetaTableRow>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <TableHeaderCell>
                    Hello
                  </TableHeaderCell>,
                  <TableHeaderCell>
                    World
                  </TableHeaderCell>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Hello",
                  },
                  "ref": null,
                  "rendered": "Hello",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "World",
                  },
                  "ref": null,
                  "rendered": "World",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <MetaTableRow>
                <MetaTableCell>
                  Foo
                </MetaTableCell>
                <MetaTableCell>
                  Bar
                </MetaTableCell>
              </MetaTableRow>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <MetaTableCell>
                    Foo
                  </MetaTableCell>,
                  <MetaTableCell>
                    Bar
                  </MetaTableCell>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Foo",
                  },
                  "ref": null,
                  "rendered": "Foo",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Bar",
                  },
                  "ref": null,
                  "rendered": "Bar",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "table",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
